tools/fs/safe-watcher.ts

```diff
@@ -155,6 +185,51 @@ function shouldIgnorePath(absPath: string): boolean {
   return false;
 }
 
+/**
+ * Check if a path is a symbolic link.
+ * 
+ * Symbolic links are not supported natively in some operating systems,
+ * so we need to use polling for them to ensure they are properly watched.
+ * This function is used to determine if a path is a symbolic link,
+ * so we can use polling instead of native watching for it.
+ * 
+ * If a path is a symbolic link, its root is added to the symlinkRoots set.
+ */
+function isSymbolicLink(absPath: string, addToRoots = true): boolean {
```

Comment:
This seems to work when watching something that is a symbolic link. It's possible though that the path being watched isn't a symbolic link, but is inside of one. For example:

- app
   - a (symbolic link to ../shared)
      - b

B is not a symbolic link so `isSymbolicLink` would return false, but since it is inside of one (it's actual path is ../../shared/b) changes will not be detected by the recursive watcher on `app`.

Reply:
Completely 🤔

Good catch, I should find a fix for it.